pkg/clusteragent/admission/controllers/webhook/controller_base.go

```diff
@@ -73,13 +76,16 @@ type Webhook interface {
 	Endpoint() string
 	// Resources returns the kubernetes resources for which the webhook should
 	// be invoked
-	Resources() []string
+	Resources() map[string][]string
```

Comment:
Done: https://github.com/DataDog/datadog-agent/pull/30715/commits/d24c224624d4522cb6d517a259e85fb48450f16f
It's because the resource group for Deployment (`'apps'`) is not the same as the one for Pods(`''`). Previously we were always using pods, so it was hard coded. In the future we can probably improve that with a struct maybe.